Clean up

to-sta opened the pull request:
<!---
Thank you for your pull request! 🚀
-->

### Contributor checklist

<!-- Please replace the empty checkboxes [] below with checked ones [x] accordingly. -->

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch
- [x] I have run the tests for the backend and frontend depending on what's needed for my changes as described in the [testing section of the contributing guide](CONTRIBUTING.md#testing)

---

### Description

I worked on cleaning up the backend to make it leaner

#### Overview

1. Renamed backend/backend to backend/core
2. Script to analyze API calls in the frontend
3. Streamline BASE_BACKEND_URL
4. Remove unused API's that are just boilerplate anyways
5. Improving DB field names



<details open>
  <summary>1. Renamed backend/backend to backend/core</summary>
<br>
The inner backend folder was renamed to core to provide more clarity and better reflect its purpose as the directory holding core Django settings.

```bash
├── frontend
|    ├── pages
|    ...
├── backend
     ├── core <- (previously backend)
```
</details>


<details open>
  <summary>2. Script to analyze API calls in the frontend</summary>
<br>
To better understand which API calls are actively used, I wrote a script that scans the frontend codebase for API patterns. Currently, it looks for three different base URLs: `localhost:8000`, `127.0.0.1:8000`, and `${BASE_BACKEND_URL}`.

```py
API_PATTERNS = [
    re.compile(r"\$\{BASE_BACKEND_URL\}/\S+"),
    re.compile(r"localhost:8000/\S+"),
    re.compile(r"127.0.0.1:8000/\S+"),
]
```

Results:

```bash
API calls found in the frontend code:
-------------------------------------
Number of API calls: 8
Details:

[
    {'useAuth.ts': ['${BASE_BACKEND_URL}/auth/sign_in/`,']},
    {'create.vue': ['127.0.0.1:8000/v1/entities/organizations/",']},
    {'index.vue': ['${BASE_BACKEND_URL}/entities/groups/`,']},
    {'create.vue': ['127.0.0.1:8000/v1/content/resources/",']},
    {'index.vue': ['${BASE_BACKEND_URL}/content/resources/`,']},
    {'event.ts': ['${BASE_BACKEND_URL}/events/events/`,']},
    {'group.ts': ['${BASE_BACKEND_URL}/entities/groups/`,']},
    {'organization.ts': ['${BASE_BACKEND_URL}/entities/organizations/`,']},
]
```
</details>


<details open>
  <summary>3. Streamline BASE_BACKEND_URL</summary>
<br>
The use of `127.0.0.1`, `localhost`, and `${BASE_BACKEND_URL}`in the frontend was inconsistent. To simplify and prepare for production, all references now use the environment variable BASE_BACKEND_URL, which can be dynamically set (e.g.,` backend.activist.org` or `api.activist.org`).

Updated Results:

```bash
[   
    {'useAuth.ts': ['${BASE_BACKEND_URL}/auth/sign_in/`,']},
    {'create.vue': ['${BASE_BACKEND_URL}/v1/entities/organizations/`,']},
    {'index.vue': ['${BASE_BACKEND_URL}/entities/groups/`,']},
    {'create.vue': ['${BASE_BACKEND_URL}/v1/content/resources/`,']},
    {'index.vue': ['${BASE_BACKEND_URL}/content/resources/`,']},
    {'event.ts': ['${BASE_BACKEND_URL}/events/events/`,']},
    {'group.ts': ['${BASE_BACKEND_URL}/entities/groups/`,']},
    {'organization.ts': ['${BASE_BACKEND_URL}/entities/organizations/`,']}
]
```
</details>


<details open>
  <summary>4. Remove unused API's that are just boilerplate anyways</summary>
 <br>
Here are the API that are currently exposed:


![before (3)](https://github.com/user-attachments/assets/bd0588db-dff6-45a5-a272-7b59337c15f9)

Compared to the APIs currently in use, a significantly larger number are exposed, many of which are boilerplate `viewsets`. 

Updated:

![afterwards](https://github.com/user-attachments/assets/cc3e61ff-fc09-40e3-9b75-0c0c65a8c16b)
</details>


<details open>
  <summary>5. Improving DB field names</summary>
<br>
Model field names within django add `_id` automatically to the column name, therefore if we are naming fields:

```python
class GroupMember(models.Model):
    group_id = models.ForeignKey(Group, on_delete=models.CASCADE, related_name="group_members")
    ...
```
The SQL query would be:

```sql
SELECT * 
FROM entities.group_member
WHERE group_id_id = 'UUID'
```
What is a bit odd right? This change will needs to be adopted by the frontend.

![example_id_id](https://github.com/user-attachments/assets/284c6650-db41-4591-b105-e6fb89d7c5f6)

</details>


### Open

- [x] pytest tests
- [x] pinia stores
 @andrewtavis my assumption would be that the pinia stores might need to be updated to refelect the changes (e.g. `groupId` -> `group`) 

### Related issue

<!--- activist prefers that pull requests be related to already open issues. -->
<!--- If applicable, please link to the issue by replacing ISSUE_NUMBER with the appropriate number below. -->
<!--- Feel free to delete this section if this does not apply. -->

- No related issue

andrewtavis commented:
Are the i18n checks still failing for you, @to-sta? Just checking if your last commit used the ignore pre-commit functionality and the checks are still problematic...

to-sta commented:
> Are the i18n checks still failing for you, @to-sta? Just checking if your last commit used the ignore pre-commit functionality and the checks are still problematic...

There are no problems. I started working on this branch before we fixed the i18n check, so I initially commented out the pre-commit hook but have now reverted that change.

andrewtavis commented:
`entities` being renamed to `communities` went through in the last commit. So I can understand the last changes:

- I'm looking for instances like `r'[a-z]*Id'` and in cases where it's in the store we'll need to remove `Id` from them as the fields have been renamed in the backend to avoid an `_id_id`?
- I'll likely need to change all instances of these, meaning in type files as well?

Btw, does this close #986? If not, what else would we need to do to finalize that? :)

Thanks for all the work here, @to-sta! Lots of great changes here like the `core` one as well 😊 And thanks for cleaning up the routes!

to-sta commented:
> `entities` being renamed to `communities` went through in the last commit. So I can understand the last changes:

Good call 👍  ! I was thinking about splitting it into:

```bash
communities
├── groups
 |          ├──  ...
├── organizations
 |           ├──  ...
```
Let me know what you think? 

> * I'm looking for instances like `r'[a-z]*Id'` and in cases where it's in the store we'll need to remove `Id` from them as the fields have been renamed in the backend to avoid an `_id_id`?
> * I'll likely need to change all instances of these, meaning in type files as well?

Ahh yeah, because of the camelCase conversion 😄 almost forgot about that. Should be the types and pinia stores, but its best to search with the regex. Maybe its used as a property in a page file as well.   

> Btw, does this close #986? If not, what else would we need to do to finalize that? :)

Yes, we can close this after we merge this PR.
> Thanks for all the work here, @to-sta! Lots of great changes here like the `core` one as well 😊 And thanks for cleaning up the routes!

andrewtavis commented:
But also upon playing around with the split a bit, how would we get around all of the circular imports? They're kind of all over the place for groups and organizations where the models, serializers and factories of each rely on the other. I'm not really seeing a good way to split them right now.

to-sta commented:
> But also upon playing around with the split a bit, how would we get around all of the circular imports? They're kind of all over the place for groups and organizations where the models, serializers and factories of each rely on the other. I'm not really seeing a good way to split them right now.

Good point @andrewtavis, let's open a new issue and discuss that over there :smile:. Because the way we do a split here, should also be a template for other splits later one. Makes sense to discuss the options a bit.

andrewtavis reviewed (approved):
Thanks for the updates here, @to-sta! Really great to have all these changes coming through to make the data transfer so much more seamless 😊😊
